src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java

```diff
@@ -199,9 +209,49 @@ protected HttpExchange() {
      *                       specified and no response body may be written.
      * @throws IOException   if the response headers have already been sent or an I/O error occurs
      * @see   HttpExchange#getResponseBody()
+     * @see   HttpExchange#CHUNKED_CONTENT
+     * @see   HttpExchange#NO_CONTENT
      */
     public abstract void sendResponseHeaders(int rCode, long responseLength) throws IOException;
 
+    /**
+     * Convenience method to send the response headers with no response body.
+     * The exchange is also closed.
+     * @param code the response code to send
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final void sendResponseHeadersNoContent(int code) throws IOException {
+        sendResponseHeaders(code, NO_CONTENT);
+        close();
+    }
+    /**
+     * Convenience method to send a chunked response. The caller must write the response
+     * body to the returned output stream and then close it.
+     * @param code the response code to send
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @return the stream to write the response to. the caller must close the stream.
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final OutputStream sendResponseHeadersChunked(int code) throws IOException {
+        sendResponseHeaders(code,CHUNKED_CONTENT);
```

Comment:
I don't like the fact that this methor returns an `OutputStream` when other `sendResponseXxx` methods return `void`. IMO it should either be renamed or the return type changed to `void`, or maybe it should simply be dropped.

Reply:
The purpose was that if you state you are sending a chunked response, you must close the output stream. By returning it - the linter will catch that it is closed, or used in a try-with-resources.

Reply:
I understand - but in this case we need a different (better?) name.

Reply:
I suggested

OutputStream sendResponseChunked(int code)

in the other related comment.